src/core/hooks/useNumericRouteParams.ts

```diff
@@ -0,0 +1,11 @@
+import { useRouter } from 'next/router';
+
+export default function useNumericRouteParams(): Record<string, number> {
+  const input = useRouter().query;
+  const output: Record<string, number> = {};
+  Object.keys(input).forEach((key: string) => {
```

Comment:
Ah, filtering is a very good idea! I will add it before I merge.

The generic approach is something I tried at first but I couldn't really come up with a useful solution. And since basically all params are numbers, I ended up creating this hook instead of a generic one.